Sources/NIOInstrumentation/BaggageContextInboundHTTPHandler.swift

```diff
@@ -21,16 +21,16 @@ public final class BaggageContextInboundHTTPHandler: ChannelInboundHandler {
     public typealias InboundOut = HTTPServerRequestPart
 
     private let instrument: Instrument
-    private var onBaggageExtracted: (BaggageContext) -> Void
+    private var onBaggageExtracted: (BaggageContextProtocol) -> Void
```

Comment:
Why the protocol? We always deal with the explicit baggage here I think, we should not encourage smuggling around a different type rather than the baggage in the channels as well bts so in NIO the stored thing should be the explicit BaggageContext as well I think